### PR TITLE
Add an option to grab a profile from kubelet to grab-profiles.sh

### PR DESCRIPTION
**Pull request by gmarek**
@jszczepkowski @mwielgus @piosz  We had to do it by hand yesterday, so I added kubelet profiling to the script.

cc people who might be interested: @dchen1107 @yujuhong @davidopp @lavalamp

### CODE REVIEW

**Comment by gmarek** [comment]
I probably should add shflags to dependencies and use them instead of getopt...

File: hack/grab-profiles.sh [hunk 1]
```diff
@@ -18,17 +18,44 @@ set -o errexit
...
+  echo "binary: $binary"
+
+  for profile in ${requested_profiles}; do
+    case ${profile} in
+      cpu)
+        go tool pprof "-pdf" "${binary}" "http://localhost:${tunnel_port}/${path}/profile" > "${output_prefix}-${profile}-profile-${timestamp}.pdf"
+        ;;
+      mem)
+        for flag in ${mem_pprof_flags}; do
+          go tool pprof "-${flag}" "-pdf" "${binary}" "http://localhost:${tunnel_port}/${path}/heap" > "${output_prefix}-${profile}-${flag}-profile-${timestamp}.pdf"
```
Comment (wojtek-t) [inline hack/grab-profiles.sh]: shouldn't flag be mem_pprof_flags?

File: hack/grab-profiles.sh [hunk 2]
```diff
@@ -159,19 +208,21 @@ for component in ${profile_components}; do
...
-done
-
+  if [[ "${component}" == "kubelet" ]]; then
+    for node in $(echo ${kubelet_addreses} | sed 's/[,;]/\n/g'); do
+      grab_profiles_from_component "${requested_profiles}" "${mem_pprof_flags}" "${binary}" "${tunnel_port}" "${path}/${node}/debug/pprof" "${output_dir}/${component}" "${timestamp}"
+    done    
+  else 
+    grab_profiles_from_component "${requested_profiles}" "${mem_pprof_flags}" "${binary}" "${tunnel_port}" "${path}" "${output_dir}/${component}" "${timestamp}"
+  fi
+done
```
Comment (wojtek-t) [inline hack/grab-profiles.sh]: Can you eliminate this strange red sign from the end of line?

File: hack/grab-profiles.sh (hunk 1 above)
Comment (gmarek) [inline hack/grab-profiles.sh]: I believe that pprof can grab only one profile at a time, so we need to iterate over "flags" (really - memory profiler operation kind) and do it one by one. I can push iteration outside of this function if you think it'll be better.

File: hack/grab-profiles.sh (hunk 2 above)
Comment (gmarek) [inline hack/grab-profiles.sh]: Done.

File: hack/grab-profiles.sh (hunk 1 above)
Comment (wojtek-t) [inline hack/grab-profiles.sh]: hmm - so can you given an example of flags that you may want to pass here and that you can't pass all of them to a single pprof call?

**Comment by pmorie** [comment]
@kubernetes/rh-cluster-infra @timothysc

File: hack/grab-profiles.sh (hunk 1 above)
Comment (gmarek) [inline hack/grab-profiles.sh]: --inuse_space, --inuse_objects

Comment (wojtek-t) [inline hack/grab-profiles.sh]: Whatever those are :)
Anyway - can you please add a comment about it (also giving this example)?

Comment (gmarek) [inline hack/grab-profiles.sh]: Done.

File: hack/grab-profiles.sh [hunk 3]
```diff
@@ -18,17 +18,48 @@ set -o errexit
...
+  echo "binary: $binary"
+
+  for profile in ${requested_profiles}; do
+    case ${profile} in
+      cpu)
+        go tool pprof "-pdf" "${binary}" "http://localhost:${tunnel_port}/${path}/profile" > "${output_prefix}-${profile}-profile-${timestamp}.pdf"
+        ;;
+      mem)
+        # There are different kinds of memory profiles that are available that
+        # had do be grabbed separately: --inuse-space, --inuse-objects,
```
Comment (wojtek-t) [inline hack/grab-profiles.sh]: s/do/to/

Comment (gmarek) [inline hack/grab-profiles.sh]: Done.

**Comment by wojtek-t** [comment]
cluster didn't started

@k8s-bot e2e test this please

**Comment by gmarek** [comment]
It's a change in script that is only run by people. All tests except linters are pointless in this case:)

**Pull request merged**